### PR TITLE
SC-46: type corner cases tests.

### PR DESCRIPTION
**Pull request by Blackmorse**
Moved for using of Strings instead of (U)Int(128+) types because spark can't handle it with a proper precision

### CODE REVIEW

File: src/it/scala/com/blackmorse/spark/clickhouse/types/BaseTestCases.scala [hunk 1]
```diff
@@ -72,7 +75,7 @@ object BaseTestCases extends should.Matchers {
             case t: clickhouseType.T => t
           }
         }
-      .sortBy(_.size)
+      .sortBy(e => (e.size, e.map(_.hashCode()).sum))
```
Comment (WonderBeat) [inline src/it/scala/com/blackmorse/spark/clickhouse/types/BaseTestCases.scala:78]: 👍🏼

**Review by WonderBeat — APPROVED** [review]
String instead of BigDecimal is nice, but hope it won't get us into trouble

**Comment by Blackmorse** [comment]
@WonderBeat , I'm not happy with this solution, hope that you have better ideas :)
Even now it is problem, because user can't push Spark's Decimal field into Clickhouse's field. Will fix it in upcoming PR's
